test/resolver.spec.js

```diff
@@ -22,282 +19,209 @@ describe('IPLD Format resolver (local)', () => {
   let dataLinksNodeBlob
 
   const links = [{
-    name: '',
-    cid: 'QmXg9Pp2ytZ14xgmQjYEiHjVjMFXzCVVEcRTWJBmLgR39U',
-    size: 10
+    Name: '',
+    Hash: new CID('QmXg9Pp2ytZ14xgmQjYEiHjVjMFXzCVVEcRTWJBmLgR39U'),
+    Tsize: 10
   }, {
-    name: 'named link',
-    cid: 'QmXg9Pp2ytZ14xgmQjYEiHjVjMFXzCVVEcRTWJBmLgR39V',
-    size: 8
+    Name: 'named link',
+    Hash: new CID('QmXg9Pp2ytZ14xgmQjYEiHjVjMFXzCVVEcRTWJBmLgR39V'),
+    Tsize: 8
   }]
-  const create = (data, links, callback) => waterfall([
-    (cb) => DAGNode.create(data, links, cb),
-    (n, cb) => utils.serialize(n, cb)
-  ], callback)
-
-  before((done) => {
-    parallel([
-      (cb) => create(Buffer.alloc(0), [], cb),
-      (cb) => create(Buffer.alloc(0), links, cb),
-      (cb) => create(Buffer.from('aaah the data'), links, cb)
-    ], (err, res) => {
-      expect(err).to.not.exist()
-      emptyNodeBlob = res[0]
-      linksNodeBlob = res[1]
-      dataLinksNodeBlob = res[2]
-      done()
-    })
-  })
-
-  it('multicodec is dag-pb', () => {
-    expect(resolver.multicodec).to.equal('dag-pb')
-  })
-
-  it('defaultHashAlg is sha2-256', () => {
-    expect(resolver.defaultHashAlg).to.equal('sha2-256')
+  const create = async (data, links) => {
```

Comment:
I should probably have stopped earlier re `async` and `await`, will put thoughts about async/await in the main comments